cpp/oneapi/dal/table/homogen.hpp

```diff
@@ -119,18 +121,39 @@ class ONEDAL_EXPORT homogen_table : public table {
     }
 #endif
 
-public:
+    /// Creates a new ``homogen_table`` instance from externally-defined data block.
+    /// The created table stores the provided array object.
```

Comment:
I'm not sure it is good to say that the table stores an array. It is up to implementation in my opinion, but the table shall store the data contained in the array.
There is no details about how it should refer to the data inside the array. `wrap()` over raw pointers does not own it, here the behavior is opposite I guess. We need to articulate this in the docs. 
Another thing is about information that the table can use from the array. Should table preserve the information about data mutability in the array? I think yes, it is needed.

Reply:
Agree with the point about what table stores.

> Another thing is about information that the table can use from the array. Should table preserve the information about data mutability in the array? I think yes, it is needed.

Tables are always immutable. Information on mutability can be preserved for some reason or not, this is implementation details.